packages/fxa-content-server/tests/intern.js

```diff
@@ -0,0 +1,162 @@
+const testPlanId = args.testPlanId || 'plan_GqM9N6qyhvxaVk';
+
+// Intern specific options are here: https://theintern.io/docs.html#Intern/4/docs/docs%2Fconfiguration.md/properties
+const config = {
+  asyncTimeout: asyncTimeout,
+  bail: bailAfterFirstFailure,
+  defaultTimeout: 45000, // 30 seconds just isn't long enough for some tests.
+  environments: {
+    browserName: 'firefox',
+    fixSessionCapabilities: 'no-detect',
+    usesHandleParameter: true,
+  },
+  filterErrorStack: true,
+
+  fxaAuthRoot: fxaAuthRoot,
+  fxaContentRoot: fxaContentRoot,
+  fxaSettingsV2Root: fxaSettingsV2Root,
+  fxaDevBox: fxaDevBox,
+  fxaEmailRoot: fxaEmailRoot,
+  fxaOAuthApp: fxaOAuthApp,
+  fxaOAuthRoot: fxaOAuthRoot,
+  fxaProduction: fxaProduction,
+  fxaProfileRoot: fxaProfileRoot,
+  fxaToken: fxaToken,
+  fxaTokenRoot: fxaTokenRoot,
+  fxaUntrustedOauthApp: fxaUntrustedOauthApp,
+  fxaPaymentsRoot,
+
+  pageLoadTimeout: 30000,
+  reporters: [
+    {
+      name: 'junit',
+      options: {
+        filename: output,
+      },
+    },
+    'runner',
+  ],
+  serverPort: 9091,
+  serverUrl: 'http://localhost:9091',
+  socketPort: 9077,
+  tunnelOptions: {
+    drivers: [
+      {
+        name: 'firefox',
+      },
+    ],
+  },
+
+  testProductId,
+  testPlanId,
+};
+
+if (args.grep) {
+  config.grep = new RegExp(args.grep, 'i');
+}
+
+if (args.suites) {
+  switch (args.suites) {
+    case 'server':
+    case 'server-resources':
+      config.functionalSuites = [];
+      config.node = {
+        suites: testsServer,
+      };
+      config.tunnelOptions = {};
+      config.environments = {
+        browserName: 'node',
+      };
+      config.reporters = 'pretty';
+      if (args.suites === 'server-resources') {
+        config.node.suites = testsServerResources;
+      }
+      break;
+  }
+}
+
+if (args.unit) {
+  config.functionalSuites.unshift('tests/functional/mocha.js');
+}
+
+config.capabilities = {};
+config.capabilities['moz:firefoxOptions'] = {};
+// to create a profile, give it the `config` option.
+config.capabilities['moz:firefoxOptions'].profile = firefoxProfile(config); //eslint-disable-line camelcase
+// uncomment to show devtools on launch
+// config.capabilities['moz:firefoxOptions'].args = ['-devtools'];
+
+// custom Firefox binary location, if specified then the default is ignored.
+// ref: https://code.google.com/p/selenium/wiki/DesiredCapabilities#WebDriver
+
+const failed = [];
+
+intern.on('suiteEnd', (test) => {
+  if (test.error) {
+    failed.push(test);
+  }
+});
+
+intern.on('testEnd', (test) => {
+  if (test.error) {
+    failed.push(test);
+  }
+});
+
+intern.on('afterRun', () => {
+  if (failed.length) {
+    // This text output is used later to target tests to rerun. The value
+    // is essentially just a big regex. Note that that the postfixed $
+    // helps the regex be more precisely.
+    fs.writeFileSync('rerun.txt', failed.map((f) => `${f.name}$`).join('|'));
```

Comment:
I don't think we need this anymore since we no longer use the intern to run the functional tests.

Reply:
But we are still using intern to run the server tests. Is this not a part of that or just for functional tests?

Reply:
We are using intern to run the server tests.  But the retries, which is what that text file is used for, were meant for the flaky browser based functional tests.